infra/templates/gitlab-bootstrap-iac-cicd/README.md

```diff
@@ -0,0 +1,199 @@
+# Bootstrap for Terraform deployments through Gitlab into Azure
+
+This directory contains [Terraform](https://www.terraform.io/) templates that can bootstrap Azure and Gitlab resources in a way that enables running robust CICD of [Terraform](https://www.terraform.io/) templates using [Gitlab CICD](https://docs.gitlab.com/ee/ci/). After applying this template, automated CI of terraform deployments should **just work**.
+
+> **Note**: This template is intended to be used alongside the CICD pipeline for infrastructure using Gitlab. More information can be found [here](../../../devops/providers/gitlab/templates/README.md)
+
+At a high level, this template aims to:
+
+* Deploy Azure Dependencies required for automated CICD of Terraform deployments
+* Configure variables in GitLab required for automated CICD of Terraform deployments
+* Configure dependencies for each a multistage (`dev`, `integration`, `prod`, etc...) Terraform deployment
+
+> **Note**: This template only sets up the **dependencies** needed to do a production ready infrastructure deployment, such as backend state, deployment credentials, Azure Contianer Reigstry and Gitlab variables.
+
+There are manythings deployed by this template, including:
+
+* Backend state storage account
+* Backend state containers for this deployment
+* ACR for storing docker images
+* GitLab variables needed for all deployments
+* For each deployment environment
+  * Backend state container
+  * Service principal used for deployments to that environment
+  * Resource group
+  * Role based security
+
+## Identities/Credentials Configured
+
+This template will generate some credentials, which are enumarated blow:
+
+| Description | Reason | Notes |
+| ---         | ---    | ---   |
+| ACR Push/Pull | Needed by the pipeline that builds the base image used by all of the infrastructure CICD in GitLab | N/A |
+| Environment Deploy | Needed by each environment to execute a deployment of resources into Azure | One generated for each environment |
+
+## Usage
+
+There are a few use cases for the code in this repository. The sections below outline the usage for each of those cases
+
+### First Time Setup
+
+Among the many resources provisioned by this template is the [Backend Configuration](https://www.terraform.io/docs/backends/index.html) that hosts the [Terraform State](https://www.terraform.io/docs/state/index.html) for this template, as well as the state for each deployment.
+
+Because of this, we'll cannot have the backend state configured for the initial deployment of this template. These steps will take you through the following:
```

Comment:
```suggestion
Because of this, we cannot have the backend state configured for the initial deployment of this template. These steps will take you through the following:
```